packages/fxa-auth-server/lib/payments/customer-locations.ts

```diff
@@ -0,0 +1,173 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+import { Logger } from 'mozlog';
+
+// Region codes from: https://en.wikipedia.org/wiki/ISO_3166-2:US and https://en.wikipedia.org/wiki/ISO_3166-2:CA
+import STATES_LONG_NAME_TO_SHORT_NAME_MAP from './states-long-name-to-short-name-map.json';
+import { StripeHelper } from './stripe';
+
+type PayPalUserLocationResult = {
+  uid: string;
+  state: string;
+  country: string;
+  count: number;
+};
+
+// The countries we need region data for
+const COUNTRIES_LONG_NAME_TO_SHORT_NAME_MAP = {
+  // The long name is used in the BigQuery metrics logs; the short name is used
+  // in the Stripe customer billing address
+  'United States': 'US',
+  Canada: 'CA',
+};
+
+function sleep(delayMs: number) {
+  return new Promise((resolve) => setTimeout(resolve, delayMs));
+}
```

Comment:
This is a utility that could be useful for other modules, but I don’t see anything similar elsewhere in the auth server. Only in fxa-metrics-processor, which I think is deprecated. Should I put this somewhere else?

Reply:
🤷 I think it's one of those things that's simple enough people just write it whenever they need it.  I'm not against putting it somewhere, but people need to know and use it.